src/jitter.rs

```diff
@@ -48,25 +48,53 @@ const MEMORY_SIZE: usize = MEMORY_BLOCKS * MEMORY_BLOCKSIZE;
 // Note: the C implementation relies on being compiled without optimizations.
 // This implementation goes through lengths to make the compiler not optimise
 // out what is technically dead code, but that does influence timing jitter.
-pub struct JitterRng {
+pub struct JitterRng<T: JitterTimer> {
     data: u64, // Actual random number
     // Number of rounds to run the entropy collector per 64 bits
-    rounds: u32,
-    // Timer and previous time stamp, used by `measure_jitter`
-    timer: fn() -> u64,
+    rounds: u8,
+    // Timer used by `measure_jitter`
+    timer: T,
+    // Memory for the Memory Access noise source FIXME
+    mem_prev_index: u16,
```

Comment:
So, is it necessary to use a different index every time when the whole block gets reallocated? I'm not really sure about the memory access bit — e.g. if the block is freshly allocated each time, what's to stop the compiler just working out the result in advance?

Reply:
Using a different index each time will keep the code working as before, I did not want to change the algorithm. Preserving the index across runs is the reason the compiler can not work out the result. But I should have updated the comment (I use a lot of `FIXME`s while making changes 😄)